### PR TITLE
AutoEnchantEvent

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
We had an AutoDisenchantEvent but not an AutoEnchantEvent :thonk:

## Changes
Created AutoEnchantEvent to go along with AutoDisenchantEvent

## Related Issues
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

---

https://user-images.githubusercontent.com/8492901/114275154-52444d80-9a19-11eb-861b-1046706dc018.mp4

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/api/events/AutoDisenchantEvent.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
  * 
  * @author poma123
  *
+ * @see AutoDisenchantEvent
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/api/events/AutoDisenchantEvent.java:18]: *glares*

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
Wrong class reference :eyes:

Also can you please add an `(API) Added AutoEnchantEvent` entry to the changelog?

**Comment by WalshyDev** [comment]
We really need a checkbox for changelog

**Comment by TheBusyBiscuit** [comment]
> We really need a checkbox for changelog

I am highly against that.
If everyone was to edit the changelog, it would just cause a massive load of merge conflicts.
In my opinion, we should update the changelog manually.
The only reason why I asked for it here is because this is an internal PR.

**Review by TheBusyBiscuit — APPROVED** [review]
lgtm

**Review by Sfiguz7 — APPROVED** [review]

**Comment by WalshyDev** [comment]
Last commit failed due to time out. It's fine. I can't merge though due to that

**Comment by TheBusyBiscuit** [comment]
Re-scheduled the action.

**Pull request merged**